numpy/_core/src/multiarray/multiarraymodule.c

```diff
@@ -1997,14 +1997,11 @@ array_empty(PyObject *NPY_UNUSED(ignored),
     ret = (PyArrayObject *)PyArray_Empty_int(
         shape.len, shape.ptr, dt_info.descr, dt_info.dtype, is_f_order);
 
-    npy_free_cache_dim_obj(shape);
-    return (PyObject *)ret;
-
 fail:
     Py_XDECREF(dt_info.descr);
     Py_XDECREF(dt_info.dtype);
     npy_free_cache_dim_obj(shape);
-    return NULL;
+    return (PyObject *)ret;
```

Comment:
This one looks odd, I assume there is some reason to return `ret`.

Reply:
`ret` is NULL initialized, so why not reduce the cleanup duplication.